sdk/core/core/inc/az_span.h

```diff
@@ -226,213 +185,186 @@ AZ_NODISCARD bool az_span_is_content_equal_ignoring_case(az_span span1, az_span
- * starting at the destination span length.
+ * @brief az_span_fill Fills all the bytes of the destination span with the specified value.
  *
- * @param[in] destination The az_span where the bytes should be appended to
- * @param[in] source The uint32 whose number is appended to the destination span as ASCII
- * digits
- * @param[out] out_span A pointer to an az_span that receives the span referring to the
- * destination span with its length updated
- * @return An #az_result value indicating the result of the operation:
+ * @param[in] destination The span whose bytes will be set to \p value.
+ * @param[in] value The byte to be replicated within the destination span.
+ */
+AZ_INLINE void az_span_fill(az_span destination, uint8_t value)
+{
+  memset(az_span_ptr(destination), value, (size_t)az_span_size(destination));
+}
+
+/******************************  SPAN PARSING AND FORMATTING */
+
+/**
+ * @brief az_span_atou64 Parses an az_span containing ASCII digits into a uint64 number.
+ *
+ * @param[in] span The az_span containing the ASCII digits to be parsed.
+ * @param[in] out_number The pointer to the variable that is to receive the number
+ * @return An #az_result value indicating the result of the operation.
  *          #AZ_OK if successful
- *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the destination is not big enough to
- * contain the appended bytes
+ *          #AZ_ERROR_PARSER_UNEXPECTED_CHAR if a non-ASCII digit is found within the span.
  */
-AZ_NODISCARD az_result
-az_span_append_u32toa(az_span destination, uint32_t source, az_span* out_span);
+
+AZ_NODISCARD az_result az_span_atou64(az_span span, uint64_t* out_number);
 
 /**
- * @brief az_span_append_i64toa appends an int64 as digit characters to the destination
- * starting at the destination span length.
+ * @brief az_span_atou Parses an az_span containing ASCII digits into a uint32 number.
  *
- * @param[in] destination The az_span where the bytes should be appended to
- * @param[in] source The int64 whose number is appended to the destination span as ASCII
- * digits
- * @param[out] out_span A pointer to an az_span that receives the span referring to the
- * destination span with its length updated
- * @return An #az_result value indicating the result of the operation:
+ * @param span The az_span containing the ASCII digits to be parsed.
+ * @param out_number The pointer to the variable that is to receive the number
+ * @return An #az_result value indicating the result of the operation.
  *          #AZ_OK if successful
- *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the destination is not big enough to
- * contain the appended bytes
+ *          #AZ_ERROR_PARSER_UNEXPECTED_CHAR if a non-ASCII digit is found within the span.
  */
-AZ_NODISCARD az_result
-az_span_append_i64toa(az_span destination, int64_t source, az_span* out_span);
+AZ_NODISCARD az_result az_span_atou(az_span span, uint32_t* out_number);
```

Comment:
You don't think this function name should end with atou32?

Reply:
Here's my rationale: I was trying to mirror existing atoi and itoa APIs in C/C++ which don't explicitly state the bit-size of the int. It is assumed that no bit-size means 32 (likely that is the most common) and for all others, the bit-size of the number is mentioned (i8, u8, i64, u64). If you think internal consistency is more valuable rather than being concise for the common case/consistent with C APIs, I am fine with adding the `32` bit-size for int/uint.

To your point, I think `atoi` makes sense to keep, but that means we would add `atou`, which is a little weird (also too close to auto with a typo). I am fine with either direction.